lib/tipg-api/index.ts

```diff
@@ -0,0 +1,112 @@
+import {
+    Stack,
+    aws_ec2 as ec2,
+    aws_rds as rds,
+    aws_lambda as lambda,
+    aws_secretsmanager as secretsmanager,
+    CfnOutput,
+    Duration,
+  } from "aws-cdk-lib";
+  import {
+    PythonFunction,
+    PythonFunctionProps,
+  } from "@aws-cdk/aws-lambda-python-alpha";
+  import { HttpApi } from "@aws-cdk/aws-apigatewayv2-alpha";
+  import { HttpLambdaIntegration } from "@aws-cdk/aws-apigatewayv2-integrations-alpha";
+  import { Construct } from "constructs";
+
+  export class TiPgApiLambda extends Construct {
+    readonly url: string;
+    public tiPgLambdaFunction: PythonFunction;
+
+    constructor(scope: Construct, id: string, props: TiPgApiLambdaProps) {
+      super(scope, id);
+
+      const apiCode = props.apiCode || {
```

Comment:
I was using the default api code. These few lines could be the culprits ? If I compare to titiler-pgstac the set up is different. We use 'assets from docker build' for titiler-pgstac, and I just checked, in the Dockerfile we explicitly transfer _all_ the python files (including `utils.py`) to the Docker filesystem. Are we sure this happens here ?

Reply:
@emileten I'm not 💯 sure I follow everything, can you point me toward the code you are using. 

For this `apiCode` I think I copy/pasted from the STAC API code...

Reply:
I need to push the changes to the wrapper I'm using @vincentsarago, haven't done it yet. I might be doing something wrong in it, but I'm just calling the new construct of this PR without passing any value to apiCode, so the default should be used.

I'll look for the cause of this problem tomorrow !

Reply:
[solved here.](https://github.com/developmentseed/eoapi-cdk/pull/62/commits/26ad3134403e79684788cbf54709d60177d261ab)

it's the same problem that was caused by https://github.com/developmentseed/eoapi-cdk/pull/61 but there it was with the tiler.

Here, the problem was : we do `from tipg.main import app` at the top of the handler. Doing this import [triggers a postgres settings creation](https://github.com/developmentseed/tipg/blob/1fc3b73c17da45826dc821414583059459e7b1b1/tipg/main.py#L28), but the postgres environment variables are _not_ yet loaded into the environment, it was happening [_after_ all imports](https://github.com/developmentseed/eoapi-cdk/blob/83304a61bcfacba3ecc292bb557a4c14fee494a5/lib/tipg-api/runtime/src/handler.py#L21). There are only two ways around this : 

1. load the postgres env vars before the imports (we don't want to have them in the lambda configuration directly, because it exposes secrets). That's [what I did here.](https://github.com/developmentseed/eoapi-cdk/pull/62/commits/26ad3134403e79684788cbf54709d60177d261ab) 
2. _don't_ rely on `from tipg.main import app` and have a custom app instead. [That's what @vincentsarago was suggesting we do here](https://github.com/developmentseed/eoapi-cdk/pull/42#discussion_r1212805079), and that's what [we do for stac-fastapi](https://github.com/developmentseed/eoapi-cdk/blob/main/lib/stac-api/runtime/src/app.py) (however I'm not sure the stac-fastapi provide an app layer that's reusable like tipg or titiler-pgstac). 

Back when I added the titiler pgstac construct here I decided to reuse the standard app layer from the titiler package (titiler.main.app) nbecause I saw no reason to duplicate code. 

Anyways, just like for the tiler bug, I chose (1) to solve this bug.